app/models/concerns/recordable.rb

```diff
@@ -13,13 +13,15 @@ module Recordable
   # Centralises the logic around deciding what points to score for an item
   #
   # Including classes should implement: a +score+ method or attribute
-  def score_when_recorded_at(school, date)
-    academic_year = school.academic_year_for(date)
+  def score_when_recorded_at(observation)
+    academic_year = observation.school.academic_year_for(observation.at)
     return nil unless academic_year&.current?
-    return nil if count_existing_for_academic_year(school, academic_year) >= maximum_frequency
+    uncounted = observation.new_record? || observation.points.to_i.zero? ? 1 : 0
```

Comment:
not understanding the requirements around having no points - is this to compensate for count_existing_for_academic_year ignoring no points? is it ok if there are existing observations with no points?

Reply:
I've added a bit more documentation in the code and changed the code to just count unsaved as observations with no points (a new record would have no points anyway).

Hopefully this helps?